src/cascadia/TerminalSettingsModel/TerminalSettings.cpp

```diff
@@ -67,10 +67,18 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         THROW_HR_IF_NULL(E_INVALIDARG, profile);
 
         const auto globals = appSettings.GlobalSettings();
-        settings->_ApplyProfileSettings(profile, globals.ColorSchemes());
+        settings->_ApplyProfileSettings(profile);
         settings->_ApplyGlobalSettings(globals);
+        settings->_ApplyAppearanceSettings(profile.DefaultAppearance(), globals.ColorSchemes());
 
-        return *settings;
+        if (profile.UnfocusedAppearance())
+        {
+            auto child = settings->CreateChild();
+            child->_ApplyAppearanceSettings(profile.UnfocusedAppearance(), globals.ColorSchemes());
+            return winrt::make<TerminalSettingsStruct>(*settings, *child);
+        }
+
+        return winrt::make<TerminalSettingsStruct>(*settings, nullptr);

```

Comment:
Consider simplifying so that you only have one return. Take advantage of the fact that you can have a local var storing `nullptr` and it's _fine_ :smile:

Reply:
Could you elaborate a little on how we can do this? I tried:

```
        winrt::com_ptr<TerminalSettings> child{ nullptr };
        if (profile.UnfocusedAppearance())
        {
            child = settings->CreateChild();
            child->_ApplyAppearanceSettings(profile.UnfocusedAppearance(), globals.ColorSchemes());
        }

        return winrt::make<TerminalSettingsCreateResult>(*settings, *child);
```
But that causes a read access violation if there's no UnfocusedAppearance. Also, we cannot initialize `child` with `TerminalSettings child{ nullptr }`

Reply:
Simplified to only have one return now!